DOCUMENTATION.md

```diff
@@ -37,61 +38,114 @@ The output tensor `ys` will have shape `(t_size, batch_size, state_size)`, and c
+
 The available solvers depends on the SDE type and the noise type.
 
 **Ito solvers**
+
 - `"euler"`: [Euler-Maruyama method](https://en.wikipedia.org/wiki/Euler%E2%80%93Maruyama_method)
-- `"milstein"`: [Milstein method](https://en.wikipedia.org/wiki/Milstein_method)
+- `"milstein"`: [Milstein method](https://en.wikipedia.org/wiki/Milstein_method).
 - `"srk"`: [Stochastic Runge-Kutta method](https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta_method_(SDE))
 
 **Stratonovich solvers**
+
 - `"euler_heun"`: [Euler-Heun method](https://infoscience.epfl.ch/record/143450/files/sde_tutorial.pdf)
 - `"heun"`: [Heun's method](https://arxiv.org/abs/1102.4401)
 - `"midpoint"`: Midpoint method
 - `"milstein"`: [Milstein method](https://en.wikipedia.org/wiki/Milstein_method)
+- `"reversible_heun"`: The reversible midpoint method, as introduced in [\[1\]](https://arxiv.org/abs/2105.13493).
+- `"adjoint_reversible_heun"`: A special method: pass this as part of `sdeint_adjoint(..., method="reversible_heun", adjoint_method="adjoint_reversible_heun")` for more accurate gradient calculations with the adjoint method, see [the notes on adjoint methods](#adjoints).
 
 Note that Milstein and SRK don't support general noise.
 
 Additionally, [gradient-free Milstein](https://infoscience.epfl.ch/record/143450/files/sde_tutorial.pdf) can be used by selecting Milstein, and then also passing in the keyword argument `sdeint(..., options=dict(grad_free=True))`.
 
+### Which solver should I use?
 
-### Providing specialised methods
-If your drift/diffusion have special structure, for example the drift and diffusion share some computations, then it may be more efficient to evaluate them together rather than alone. As such, if the following methods are present on `sde`, then they will be used if possible: `g_prod(t, y, v)`, `f_and_g(t, y)`, `f_and_g_prod(t, y, v)`. Here `g_prod` is expected to compute the batch matrix-vector product between the diffusion and the vector `v`. `f_and_*` should return a 2-tuple of `f(t, y)` and `g(t, y)`/`g_prod(t, y, v)` as appropriate.
+If calculating an Ito SDE, then `srk` will generally produce a more accurate estimate of the solution than `milstein`, which will generally produce a more accurate estimate than `euler`. Meanwhile `srk` is the most expensive, followed by `milstein`, followed by `euler` as the computationally cheapest.
 
-Depending on the integration method used it may suffice to provide only some of these methods (`f` and `g` are not mandatory).
+If calculating a Stratonovich SDE, then `midpoint` , `heun` and `milstein` are more computationally expensive. `euler_heun` and `reversible_heun` are the cheapest.
 
-## Adjoints
+If training neural SDEs _without_ the adjoint method then accurate SDE solutions usually aren't super important - the vector fields can learn to work with the discretisation chosen - whilst computational cost often matters a lot. This makes `euler` (for Ito) or `reversible_heun` (for Stratonovich) good choices.
```

Comment:
This is only true if training and evaluation are expected to have the same discretization. It may be the case that during evaluation time, we don't exactly want to use the same mesh size, say for reasons like saving compute or new data that's irregularly sampled.

Reply:
Just to note, irregularly sampled data doesn't really change anything. Just reparameterise the SDE onto a regular grid from the get-go.